cmd/capibmadm/cmd/powervs/port/create.go

```diff
@@ -0,0 +1,139 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package port
+
+import (
+	"context"
+	"fmt"
+	client "github.com/IBM-Cloud/power-go-client/clients/instance"
+	"github.com/IBM-Cloud/power-go-client/power/models"
+	"github.com/spf13/cobra"
+	"k8s.io/klog/v2"
+	"os"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/clients/iam"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/clients/powervs"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/options"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/printer"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/utils"
+	logf "sigs.k8s.io/cluster-api/cmd/clusterctl/log"
+	"strings"
+)
+
+type portCreateOptions struct {
+	networkID   string
+	ipAddress   string
+	description string
+}
+
+func CreateCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create PORT_NUMBER",
+		Short: "Create PowerVS Port",
+		Example: `
+# Create PowerVS port, provided networkID and IPAddress
+export IBMCLOUD_API_KEY=<api-key>
+capibmadm powervs port create --network <network_id> --ip-address <10.10.0.1> --description <description> --service-instance-id <service-instance-id> --zone <zone>`,
+	}
+
+	var portCreateOption portCreateOptions
+	cmd.Flags().StringVar(&portCreateOption.networkID, "network", "", "Network ID/ name on which port is to be created (Required)")
+	cmd.Flags().StringVar(&portCreateOption.ipAddress, "ip-address", "", "IP Address to be assigned to the port")
+	cmd.Flags().StringVar(&portCreateOption.description, "description", "", "Description of the port")
+	cmd.RunE = func(cmd *cobra.Command, args []string) error {
+		if err := createPort(cmd.Context(), portCreateOption); err != nil {
+			return err
+		}
+		return nil
+	}
+	options.AddCommonFlags(cmd)
+	_ = cmd.MarkFlagRequired("network")
+	return cmd
+}
+
+func createPort(ctx context.Context, portCreateOption portCreateOptions) error {
+	log := logf.Log
+	log.Info("Creating Port ", "Network ID/Name", portCreateOption.networkID, "IP Address", portCreateOption.ipAddress, "Description", portCreateOption.description, "service-instance-id", options.GlobalOptions.ServiceInstanceID, "zone", options.GlobalOptions.PowerVSZone)
+	accountID, err := utils.GetAccountID(ctx, iam.GetIAMAuth())
+	if err != nil {
+		return err
+	}
+	session, err := powervs.NewPISession(accountID, options.GlobalOptions.PowerVSZone, options.GlobalOptions.Debug)
+	if err != nil {
+		return err
+	}
+
+	networkClient := client.NewIBMPINetworkClient(ctx, session, options.GlobalOptions.ServiceInstanceID)
+	networks, err := networkClient.GetAll()
+	if err != nil {
+		return err
+	}
+	var networkNames, networkIDs []string
```

Comment:
Can follow the logic [here](https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/pull/1084) to find the netID.
No need of list and Contains func.